crates/provider/src/utils.rs

```diff
@@ -52,6 +57,62 @@ pub fn eip1559_default_estimator(
     Eip1559Estimation { max_fee_per_gas: potential_max_fee, max_priority_fee_per_gas }
 }
 
+/// Extracts the authorization information from the given URL.
+pub fn extract_auth_info(url: Url) -> Option<Authorization> {
+    if url.has_authority() {
+        let username = url.username();
+        let pass = url.password().unwrap_or_default();
+        Some(Authorization::basic(username, pass))
+    } else {
+        None
+    }
+}
+
+/// Identifies the intended transport type from the given string.
+pub fn parse_str_to_tranport_type(s: &str) -> Result<String, RpcError<TransportErrorKind>> {
+    if s.parse::<SocketAddr>().is_ok() {
+        return Ok(format!("http://{}", s));
+    }
+    // Check if s is a path and it exists
+    let path = Path::new(&s);
+    if path.is_file() {
+        // IPC if it exists
+        return Ok(s.to_string());
+    }
+
+    // Parse the URL or return an error
+    Url::parse(s).map_err(TransportErrorKind::custom)?;
+
+    Ok(s.to_string())
+}
```

Comment:
what does this do?

this function seems redundant, should be integrated in fromstr